parquet/src/test/java/org/apache/iceberg/parquet/TestBloomRowGroupFilter.java

```diff
@@ -0,0 +1,950 @@
+          in("id", INT_MIN_VALUE + i, INT_MIN_VALUE + 3 * i)).shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertTrue("Should read: integer within range", shouldRead);
+    }
+
+    for (int i = 1; i <= INT_VALUE_COUNT; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA,
+          in("id", INT_MIN_VALUE - i, INT_MIN_VALUE - 3 * i)).shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertFalse("Should not read: integer outside range", shouldRead);
+    }
+
+    for (int i = 1; i <= INT_VALUE_COUNT; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA,
+          in("id", INT_MAX_VALUE + i, INT_MAX_VALUE + 3 * i)).shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertFalse("Should not read: integer outside range", shouldRead);
+    }
+
+    boolean shouldRead = new ParquetBloomRowGroupFilter(
+        SCHEMA,
+        in("id", IntStream.range(INT_MIN_VALUE - 10, INT_MAX_VALUE + 10).boxed().collect(Collectors.toList()))
+    ).shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertTrue("Should read: the bloom is a subset of the in set", shouldRead);
+
+    shouldRead = new ParquetBloomRowGroupFilter(
+        SCHEMA,
+        in("id", IntStream.range(INT_MIN_VALUE, INT_MAX_VALUE + 1).boxed().collect(Collectors.toList()))
+    ).shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertTrue("Should read: the bloom is equal to the in set", shouldRead);
+  }
+
+  @Test
+  public void testOtherTypesIn() {
+    boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, in("all_nulls", 1, 2))
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertFalse("Should not read: in on all nulls column (bloom is empty) ", shouldRead);
+
+    shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, in("some_nulls", "aaa", "some"))
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertTrue("Should read: in on some nulls column", shouldRead);
+
+    shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, in("some_nulls", "aaa", "bbb"))
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertFalse("Should not read: some_nulls values are not within the set", shouldRead);
+
+    shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, in("no_nulls", "aaa", "bbb"))
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertFalse("Should not read: in on no nulls column (empty string is not within the set)", shouldRead);
+
+    shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, in("no_nulls", "aaa", ""))
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertTrue("Should read: in on no nulls column (empty string is within the set)", shouldRead);
+  }
+
+  @Test
+  public void testIntegerNotIn() {
+    for (int i = 0; i < INT_VALUE_COUNT; i++) {
```

Comment:
Here as well, I think you should test the same cases as `in`.

Reply:
Changed. Thanks!